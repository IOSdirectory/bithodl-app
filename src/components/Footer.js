import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import { ifIphoneX } from 'react-native-iphone-x-helper';

const styles = StyleSheet.create({
  footer: {
    alignItems: 'center',
    backgroundColor: '#ffffff',
    position: 'absolute',
    bottom: ifIphoneX(54, 30),
    left: 40,
    right: 40
  }
});

export default class Footer extends Component {
  render() {
    return (
      <View style={[styles.footer, this.props.style]} pointerEvents={this.props.pointerEvents}>
        {this.props.children}
      </View>
    );
  }
}

Footer.propTypes = {
  style: PropTypes.any,
  pointerEvents: PropTypes.string,
  children: PropTypes.node
};
