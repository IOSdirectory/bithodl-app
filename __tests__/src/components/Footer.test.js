import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Footer from '../../../src/components/Footer';

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Footer>
        <Text>0750eb7b-6bc0-4142-82e2-035068d50726</Text>
      </Footer>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
