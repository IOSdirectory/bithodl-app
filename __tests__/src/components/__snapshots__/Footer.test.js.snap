// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<View
  pointerEvents={undefined}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "bottom": 30,
        "left": 40,
        "position": "absolute",
        "right": 40,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
    0750eb7b-6bc0-4142-82e2-035068d50726
  </Text>
</View>
`;
